node/src/chain/header_validation.rs

```diff
@@ -93,6 +97,27 @@ impl<'a, DB: database::DB> Validator<'a, DB> {
             ));
         }
 
+        // Ensure rule of minimum block time is addressed
+        if candidate_block.timestamp
+            < self.prev_header.timestamp + MINIMUM_BLOCK_TIME
+        {
+            return Err(anyhow!("block time is less than minimum block time"));
+        }
+
+        let local_time = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map(|n| n.as_secs())
+            .expect("valid unix epoch");
+
+        if candidate_block.timestamp
+            > local_time + MINIMUM_BLOCK_TIME + MARGIN_TIMESTAMP
```

Comment:
I don't think we agreed on any maximum time by which the block should be received. We mentioned a margin of error wrt to our local time (since we might be slightly off sync), but that relates to the condition of the block timestamp not being in the future for us.

Reply:
This check should simply be:
`local_time > candidate_block.timestamp - MARGIN_TIMESTAMP`
and `MARGIN_TIMESTAMP` should be something smaller, like 2-3 seconds

Reply:
Should it be `candidate_block.timestamp > local_time  + MARGIN_TIMESTAMP`?